### ISSUE TITLE
Question on the conditional generation experiments

### ISSUE THREAD

**Issue by lateralthinker00** [david3684/flm#8]
Hello, I am looking to reproduce the conditional generation experiments. 
Are there implementation details or a code snippet for this?

Specifically, I am wondering about details regarding 
- how timestep embedding was handled for conditioning vs non-conditioning inputs, - any details on embedding networks to handle multiple time values per sequence
- details on attention mask values

Thank you very much for the great work!

**Comment by david3684** [comment]
Hi,
For the conditional experiment in the paper, the prefix length and position are fixed, so it isn't strictly necessary to use separated time conditions for the conditioning positions. Therefore we used a single global t across the sequence, and excluded the prefix positions from the training loss. For multi-step generation, we clamped the prefix back to the clean tokens at every Euler ODE step. For attention masks, we use global bidirectional attention for fixed 1024 length sequence.
For a more general implementation, there are two cleaner options,
1) Feed t=1 for the conditioning (prefix) positions and sampled t for the denoising positions, so the timestep embedding reflects that the prefix is already clean.
2) Add a separate binary indicator for prefix-conditioned positions and concatenate it channel-wise with an additional projection; we implemented the Sudoku solving experiments this way both for uniform diffusion and ours.

Hope this helps!

**Comment by lateralthinker00** [comment]
@david3684 Thank you very much!
